commands/ls.go

```diff
@@ -67,3 +80,48 @@ func cmdLs(c *cli.Context) {
 
 	w.Flush()
 }
+
+func parseFilters(filters []string) (options FilterOptions, err error) {
+	options = FilterOptions{}
+	for _, f := range filters {
+		kv := strings.SplitN(f, "=", 2)
+		key, value := kv[0], kv[1]
+
+		switch key {
+		case "swarm":
+			options.SwarmName = value
+		case "driver":
+			options.DriverName = value
+		default:
+			err = fmt.Errorf("Unsupported filter key '%s'", key)
+		}
+	}
+	return
+}
+
+func filterHosts(hosts []*libmachine.Host, filters FilterOptions) (filteredHosts []*libmachine.Host) {
+	swarmMasters := make(map[string]string)
+	for _, h := range hosts {
+		swarmOptions := h.HostOptions.SwarmOptions
+		if swarmOptions != nil && swarmOptions.Master {
+			swarmMasters[swarmOptions.Discovery] = h.Name
+		}
+	}
+
+	for _, h := range hosts {
+		if filters.SwarmName != "" {
+			if h.HostOptions.SwarmOptions != nil {
+				if filters.SwarmName == swarmMasters[h.HostOptions.SwarmOptions.Discovery] {
+					filteredHosts = append(filteredHosts, h)
+				}
+			}
+		} else if filters.DriverName != "" {
+			if h.DriverName == filters.DriverName {
+				filteredHosts = append(filteredHosts, h)
+			}
+		} else {
+			filteredHosts = append(filteredHosts, h)
+		}
+	}
+	return
```

Comment:
I was using a "naked return" (so, equivalent to `return filteredHosts`), but I see that's not very clear. I'll make it explicit :)